tests/ui/async-await/track-caller/async-block.stderr

```diff
@@ -1,12 +1,12 @@
-error[E0739]: attribute should be applied to a function definition
+error[E0658]: `#[track_caller]` on closures is currently unstable
   --> $DIR/async-block.rs:6:13
    |
-LL |       let _ = #[track_caller] async {
-   |  _____________^^^^^^^^^^^^^^^_-
-LL | |
-LL | |     };
-   | |_____- not a function definition
+LL |     let _ = #[track_caller] async {
+   |             ^^^^^^^^^^^^^^^
+   |
+   = note: see issue #87417 <https://github.com/rust-lang/rust/issues/87417> for more information
+   = help: add `#![feature(closure_track_caller)]` to the crate attributes to enable
```

Comment:
This probably should be gated behind a different feature gate, like `async_block_track_caller`.

Reply:
I was taking a look at this, and I believe it might require some effort to actually split these two apart.

The original `closure_track_caller` feature was added to both closures and generators in https://github.com/rust-lang/rust/pull/87064, and the same feature gate is also being reused for async fn as well.

Internally everything boils down to generator closures. It should be possible to special case this somewhere, but TBH I would rather punt on this and defer the discussion to the broader scope when moving to stabilize the feature, as @nbdd0121 suggested up above: https://github.com/rust-lang/rust/pull/104833#issuecomment-1413809524